test/integration/deployment/util.go

```diff
@@ -87,6 +94,40 @@ func newDeployment(name, ns string, replicas int32) *v1beta1.Deployment {
 	}
 }
 
+// newDeployment returns a RollingUpdate Deployment with with a fake container image
+func newAppsV1beta2Deployment(name, ns string, replicas int32) *v1beta2.Deployment {
```

Comment:
Thanks for enumerating the steps. This is very important - I didn't know that it was possible to have an integration test without controllers and informers, and was not aware at all the version difference could be resolved by the steps.